### PR TITLE
FAI-11599 | GitHub labels stream (source + converter) and PR labels

### PR DESCRIPTION
**Pull request by dbruno21**
## Description

GitHub labels stream (source + converter) and PR labels

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/github-source/src/streams/faros_labels.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export class FarosLabels extends StreamWithRepoSlices {
+  getJsonSchema(): Dictionary<any, string> {
+    return require('../../resources/schemas/farosLabels.json');
+  }
+
+  get primaryKey(): StreamKey {
+    return ['org', 'repo', 'name'];
+  }
+
+  get supportsIncremental(): boolean {
```
Comment (chalenge) [inline sources/github-source/src/streams/faros_labels.ts:17]: not seeing any incremental support

File: destinations/airbyte-faros-destination/src/converters/github/faros_pull_requests.ts [hunk 2]
```diff
@@ -57,12 +58,16 @@ export class FarosPullRequests extends GitHubConverter {
     // since we should always have branches for a PR
     const branchInfo = omitBy({sourceBranch, targetBranch}, isNil);
 
+    const prKey = {
+      repository: repoKey(pr.org, pr.repo, this.streamName.source),
```
Comment (chalenge) [inline destinations/airbyte-faros-destination/src/converters/github/faros_pull_requests.ts:62]: let's move repoKey to a common for GitHub converter

File: faros-airbyte-common/resources/github/queries/pull-requests-query.gql [hunk 3]
```diff
@@ -44,6 +44,15 @@ query pullRequests($owner: String!, $repo: String!, $cursor: String, $page_size:
                 }
                 deletions
                 isDraft
+                labels(first: $nested_page_size) {
```
Comment (chalenge) [inline faros-airbyte-common/resources/github/queries/pull-requests-query.gql:47]: how are we handling when there's more than 100 labels?

File: destinations/airbyte-faros-destination/src/converters/github/faros_pull_requests.ts (hunk 2 above)
Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/github/faros_pull_requests.ts]: done, had to delete useless uid from RepoKey type already present on common

File: sources/github-source/src/streams/faros_labels.ts (hunk 1 above)
Comment (dbruno21) [reply sources/github-source/src/streams/faros_labels.ts:17]: This is to keep parity to feeds where it's a incremental model. Can't support with the API (can't sort / filter by updatedAt)

**Comment by dbruno21** [comment]
PTAL

File: sources/github-source/src/streams/faros_labels.ts (hunk 1 above)
Comment (matiaslcoulougian) [reply sources/github-source/src/streams/faros_labels.ts:17]: I think we should remove it. We are always pulling all labels

**Review by matiaslcoulougian — APPROVED** [review]
Remove unused imports marked by sonarcloud. Then LGTM🚀

File: destinations/airbyte-faros-destination/src/converters/phabricator/commits.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ export class Commits extends PhabricatorConverter {
     } catch (e: any) {
       ctx.logger.warn(
         `Failed to parse revision id from commit sha: ${sha}, repository: ${
-          repository.uid
+          repository.name
```
Comment (chalenge) [inline destinations/airbyte-faros-destination/src/converters/phabricator/commits.ts:39]: why are we changing for the `phabricator` converter?

File: sources/github-source/src/streams/faros_labels.ts (hunk 1 above)
Comment (chalenge) [reply sources/github-source/src/streams/faros_labels.ts:17]: "parity to feeds": That is not what we want. We should build it fresh with the correct design not copying what we did in feeds to fit that architecture

File: destinations/airbyte-faros-destination/src/converters/bitbucket-server/common.ts [hunk 5]
```diff
@@ -33,7 +33,6 @@ export abstract class BitbucketServerConverter extends Converter {
 
   protected vcsRepoKey(projectKey: string, repoSlug: string): RepoKey {
     return {
-      uid: repoSlug.toLowerCase(),
```
Comment (chalenge) [inline destinations/airbyte-faros-destination/src/converters/bitbucket-server/common.ts:36]: before we remove these, do we know why they're there?

**Review by chalenge — COMMENTED** [review]
Are we sure we not using the repo uid we are removing?

File: destinations/airbyte-faros-destination/src/converters/bitbucket-server/common.ts (hunk 5 above)
Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/bitbucket-server/common.ts]: reverted

File: destinations/airbyte-faros-destination/src/converters/phabricator/commits.ts (hunk 4 above)
Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/phabricator/commits.ts]: reverted

**Pull request merged**